### ISSUE TITLE
Operator resets service type from LoadBalancer to ClusterIP

### ISSUE THREAD

**Issue by zposloncec** [OT-CONTAINER-KIT/redis-operator#180]
Redis operator 0.9.0 resets service type from LoadBalancer to ClusterIP after upgrade

```
3m1s        Normal    Type                     service/tsdb-redis-leader                     ClusterIP -> LoadBalancer
113s        Normal    EnsuringLoadBalancer     service/tsdb-redis-leader                     Ensuring load balancer
99s         Normal    EnsuredLoadBalancer      service/tsdb-redis-leader                     Ensured load balancer
59s         Normal    Type                     service/tsdb-redis-leader                     LoadBalancer -> ClusterIP
59s         Normal    DeletingLoadBalancer     service/tsdb-redis-leader                     Deleting load balancer
36s         Normal    DeletedLoadBalancer      service/tsdb-redis-leader                     Deleted load balancer
```

```
2021-11-30T13:48:40.804Z        INFO    controller_redis        Syncing Redis service with defined properties   {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-leader"}
2021-11-30T13:48:40.836Z        INFO    controller_redis        Redis service updation is successful    {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-leader"}
2021-11-30T13:48:40.850Z        INFO    controller_redis        Redis statefulset get action was successful     {"Request.StateFulSet.Namespace": "default", "Request.StateFulSet.Name": "tsdb-redis-follower"}
2021-11-30T13:48:40.866Z        INFO    controller_redis        Redis stateful  {"Request.StateFulSet.Namespace": "default", "Request.StateFulSet.Name": "tsdb-redis-follower"}
2021-11-30T13:48:40.872Z        INFO    controller_redis        Redis service get action is successful  {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-follower-headless"}
2021-11-30T13:48:40.874Z        INFO    controller_redis        Syncing Redis service with defined properties   {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-follower-headless"}
2021-11-30T13:48:40.886Z        INFO    controller_redis        Redis service updation is successful    {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-follower-headless"}
2021-11-30T13:48:40.931Z        INFO    controller_redis        Redis service get action is successful  {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-follower"}
2021-11-30T13:48:40.933Z        INFO    controller_redis        Syncing Redis service with defined properties   {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-follower"}
2021-11-30T13:48:40.945Z        INFO    controller_redis        Redis service updation is successful    {"Request.Service.Namespace": "default", "Request.Service.Name": "tsdb-redis-follower"}
2021-11-30T13:48:40.954Z        INFO    controller_redis        Redis statefulset get action was successful     {"Request.StateFulSet.Namespace": "default", "Request.StateFulSet.Name": "tsdb-redis-leader"}
2021-11-30T13:48:40.960Z        INFO    controller_redis        Redis statefulset get action was successful     {"Request.StateFulSet.Namespace": "default", "Request.StateFulSet.Name": "tsdb-redis-follower"}
```

I can see that serviceType was removed from the cluster definition in 0.9.0

**Comment by iamabhishek-dubey** [comment]
https://github.com/OT-CONTAINER-KIT/helm-charts/blob/main/charts/redis-cluster/values.yaml 

Please use the helm chart for creating an external service or see the external service example here. 
https://github.com/OT-CONTAINER-KIT/redis-operator/tree/master/example/external_service

**Comment by zposloncec** [comment]
We were running redis-operator 0.8.0 with OLM and multiple redis clusters with external services. After OLM automatically updated the operator to 0.9.0, the operator changed the LoadBalancer service type to CluserIP. Even after we patched the type to LB it reverted to ClusterIP.

We've fixed the issue by running redis-operator with manifest and changed redis-operator to use latest image 0.10.0 [image_link](https://quay.io/repository/opstree/redis-operator?tag=v0.10.0&tab=tags)
